plugins/glustershd/rest.go

```diff
@@ -201,3 +203,95 @@ func selfHealHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
 }
+
+func splitBrainOperationHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	volname := mux.Vars(r)["volname"]
+	operation := mux.Vars(r)["operation"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	var req glustershdapi.SplitBrainReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrJSONParsingFailed)
+		return
+	}
+
+	// check if hostname is in form of uuid, if yes, convert uuid to respective ip addr
+	if uuid.Parse(req.HostName) != nil {
+		if peers, err := peer.GetPeer(req.HostName); err == nil {
+			req.HostName = strings.Split(peers.PeerAddresses[0], ":")[0]
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "couldn't find ip for the peerid")
+			return
+		}
+	}
+
+	// Validate volume existence
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+
+	// Validate volume type
+	if !isVolReplicate(volinfo.Type) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "invalid operation for this volume type")
```

Comment:
Can you change the string to something like this "invalid operation: the volume is not a replicate volume"